tardis/io/atomic.py

```diff
@@ -310,149 +145,69 @@ def from_hdf5(cls, fname=None):
-            self.has_macro_atom = True
-            self.macro_atom_data_all = DataFrame(macro_atom_data[0].__array__())
-            self.macro_atom_references_all = DataFrame(macro_atom_data[1].__array__())
-        else:
-            self.has_macro_atom = False
-
-        if ion_cx_data is not None:
-            self.has_ion_cx_data = True
-            #TODO:Farm a panda here
-            self.ion_cx_th_data = DataFrame(np.array(ion_cx_data[0]))
-            self.ion_cx_th_data.set_index(['atomic_number', 'ion_number', 'level_id'], inplace=True)
-
-            self.ion_cx_sp_data = DataFrame(np.array(ion_cx_data[1]))
-            self.ion_cx_sp_data.set_index(['atomic_number', 'ion_number', 'level_id'])
-        else:
-            self.has_ion_cx_data = False
-
-        if zeta_data is not None:
-            self.zeta_data = zeta_data
-            self.has_zeta_data = True
-        else:
-            self.has_zeta_data = False
-
-        if collision_data[0] is not None:
-            self.collision_data = DataFrame(collision_data[0])
-            self.collision_data_temperatures = collision_data[1]
-            self.collision_data.set_index(['atomic_number', 'ion_number', 'level_number_lower', 'level_number_upper'],
-                                          inplace=True)
-
-            self.has_collision_data = True
-        else:
-            self.has_collision_data = False
-
-        if synpp_refs is not None:
-            self.has_synpp_refs = True
-            self.synpp_refs = pd.DataFrame(synpp_refs)
-            self.synpp_refs.set_index(['atomic_number', 'ion_number'], inplace=True)
-
-        else:
-            self.has_synpp_refs = False
-
-        self.atom_data = DataFrame(atom_data.__array__())
-        self.atom_data.set_index('atomic_number', inplace=True)
-        # We have to use constants.u because astropy uses different values for the unit u and the constant.
-        # This is changed in later versions of astropy (the value of constants.u is used in all cases)
-        if units.u.cgs == constants.u.cgs:
-            self.atom_data.mass = units.Quantity(self.atom_data.mass.values, 'u').cgs
-        else:
-            self.atom_data.mass = constants.u.cgs * self.atom_data.mass.values
-
-        self.ionization_data = DataFrame(ionization_data.__array__())
-        self.ionization_data.set_index(['atomic_number', 'ion_number'], inplace=True)
-        self.ionization_data.ionization_energy = units.Quantity(self.ionization_data.ionization_energy.values, 'eV').cgs
-
-        self.levels = DataFrame(levels_data.__array__())
-        self.levels.energy = units.Quantity(self.levels.energy.values, 'eV').cgs
-
-        self.lines = DataFrame(lines_data.__array__())
-        self.lines['nu'] = units.Quantity(self.lines['wavelength'], 'angstrom').to('Hz', units.spectral())
-        self.lines['wavelength_cm'] = units.Quantity(self.lines['wavelength'], 'angstrom').cgs
-
-
+        self.atom_data = atom_data
```

Comment:
@wkerzendorf I have the following check in mind: there is a list of related attributes:
```
    related_attrs = [("macro_atom_data", "macro_atom_references"),
                     ("collision_data", "collision_data_temperatures")]
```
It is checked for each tuple from the list if all attributes from the tuple are `None` or if all of them are not `None`. If any or the attributes is `None` while there are attributes that are not `None`, then `AtomDataMissingError` is raised with a message informing the user about  the related attributes that weren't provided. Does it make sense?